core/kernel/notif_default.c

```diff
@@ -87,7 +87,7 @@ void notif_send_async(uint32_t value, uint16_t guest_id __maybe_unused)
 	uint32_t old_itr_status = 0;
 	struct itr_chip *itr_chip = interrupt_get_main_chip();
 
-	assert(value <= NOTIF_ASYNC_VALUE_MAX && !guest_id);
+	assert(value <= NOTIF_ASYNC_VALUE_MAX);
```

Comment:
This will not work as intended for `guest_id != 0`, the `notif_values` bitstring must be per guest.
For the FF-A configuration, we have a separate implementation (in `core/arch/arm/kernel/thread_spmc.c`) due to how it must interact with the FF-A framework.

Reply:
Is there a way around this? Because, test_wd_callback() does pass guest_id>0 here.

Reply:
I don't think Xen supports notifications via the SMC ABI. That's why you're running into this problem.
Xen does support notifications via the FF-A ABI, but I assume you're using the SMC ABI for a reason.
I can try to write some code to support this with the SMC ABI, but I don't have a way of testing it so I need your help.

Reply:
Sure

Reply:
@jenswi-linaro please check the new PR, where I have made the suggested changes for the other two commits.

Sorry for the duplicate, made some unresolvable changes on local tree.

Reply:
> I don't think Xen supports notifications via the SMC ABI. That's why you're running into this problem. Xen does support notifications via the FF-A ABI, but I assume you're using the SMC ABI for a reason. I can try to write some code to support this with the SMC ABI, but I don't have a way of testing it so I need your help.

Is is possible to init notif_values (or any other entity personal to a guest) into the TEE part? So its replicated for each guest?

Reply:
Yes, I'll post a link in the new PR.

Reply:
Sure